### ISSUE TITLE
Feature request: Only fetch from GitHub when rev is updated

### ISSUE THREAD

**Issue by LeOtaku** [nmattia/niv#23]
`niv update` currently always downloads and unpacks the tarballs received from GitHub, even if the revision has not changed. For a project that has multiple larger dependencies, this can be very time-consuming. (especially since the tarballs are fetched 2 times, once for the initial update and once at the following build)

As Niv already has GitHub support in updating the rev and sha based on the `branch` property, maybe you would be willing to add a special case where the tarball is not refetched should the rev stay the same.

I personally think this would be a good improvement to the UX of Niv, but it might of course not fit with your vision for the project.

**Comment by nmattia** [comment]
Great idea! What do you think of #24?

**Comment by nmattia** [comment]
And FYI I've got a branch that fetches the updates concurrently: https://github.com/nmattia/niv/compare/nm-wip

**Comment by LeOtaku** [comment]
I've now tested both the `nm-update-on-changed` as well as the `nm-wip` branches as well as a merge of the two. For what little I have tested, all of them seem to work perfectly.

I unfortunately cannot comment on the correctness of the actual code, as I my knowledge of Haskell is very limited.

Thank you very much for responding to my request this quickly! I will keep this issue open until you decide to/not to merge the branch into master.
